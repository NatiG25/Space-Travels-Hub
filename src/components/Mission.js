import React from 'react';
import PropTypes from 'prop-types';

const Mission = ({ missionName, description }) => {
  Mission.propTypes = {
    missionName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  };

  return (
    <main>
      <section className="missionsSection">
        <h2>Mission</h2>
        <p>{missionName}</p>
      </section>
      <section className="descriptionSection">
        <h2>Description</h2>
        <p>{description}</p>
      </section>
      {/* <section className="statusSection">
      <h2>Status</h2>
    </section> */}
    </main>
  );
};

export default Mission;
