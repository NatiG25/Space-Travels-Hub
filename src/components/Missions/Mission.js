import React from 'react';
import PropTypes from 'prop-types';

const Mission = ({
  missionName,
  description,
  joinMissionHandler,
  cancelMissionHandler,
  reservation,
}) => {
  Mission.propTypes = {
    missionName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reservation: PropTypes.bool,
    joinMissionHandler: PropTypes.func.isRequired,
    cancelMissionHandler: PropTypes.func.isRequired,
  };

  Mission.defaultProps = {
    reservation: false,
  };

  return (
    <>
      <tr className="mission-row">
        <td>{missionName}</td>
        <td>{description}</td>
        <td className="status">
          {reservation
            ? <span className="status-field available-status">Active Member</span>

            : <span className="status-field unavailable-status">Not a Member</span> }
        </td>
        <td>
          {reservation
            ? (
              <button
                type="button"
                className="button-mission leave-mission"
                href="."
                onClick={cancelMissionHandler}
              >
                Cancel Mission
              </button>
            )
            : (
              <button
                type="button"
                className="button-mission join-mission"
                href="."
                onClick={joinMissionHandler}
              >
                Join Mission
              </button>
            )}
        </td>
      </tr>
    </>
  );
};

export default Mission;
