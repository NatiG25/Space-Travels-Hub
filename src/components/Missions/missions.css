.tableMissions,
th,
td {
  border: 1px solid gray;
  border-collapse: collapse;
}

.mission-row {
  background-color: lightgrey;
}

.mission-row:nth-child(even) {
  background-color: #fff;
}

.status {
  width: 12%;
  text-align: center;
}

.status-field {
  font-size: 0.8em;
  color: #fff;
  padding: 5px 10px;
  border-radius: 4px;
  width: 800px;
}

.unavailable-status {
  background: gray;
}

.available-status {
  background: #1b90a8;
}

table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 95%;
  margin: 30px auto;
}

td {
  border: 1px solid #ddd;
  text-align: left;
  padding: 8px;
}

th {
  background-color: #fff;
  border: 1px solid #ddd;
  text-align: left;
  padding: 8px;
}

.button-mission {
  padding: 5px;
  background: transparent;
  border-radius: 4px;
  cursor: pointer;
  width: 7rem;
}

.join-mission {
  border: 1px solid gray;
  color: gray;
}

.leave-mission {
  border: 1px solid #cd142d;
  color: #cd142d;
}
