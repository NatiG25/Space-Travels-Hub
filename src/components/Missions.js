import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Mission from './Mission';
import { joinMission, cancelMission } from './redux/missions/missions';

const Missions = () => {
  const missionInfo = useSelector((state) => state.missionReducer);
  const dispatch = useDispatch();

  const joinMissionHandler = (id) => {
    dispatch(joinMission(id));
  };

  const cancelMissionHandler = (id) => {
    dispatch(cancelMission(id));
  };

  return (
    <main>
      <table className="tableMissions">
        <tbody>
          <tr>
            <th>Missions</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
          {missionInfo.map((item) => (
            <Mission
              key={item.mission_id}
              missionName={item.mission_name}
              description={item.description}
              reservation={item.reserved}
              joinMissionHandler={() => joinMissionHandler(item.mission_id)}
              cancelMissionHandler={() => cancelMissionHandler(item.mission_id)}
            />
          ))}
        </tbody>
      </table>
    </main>
  );
};

export default Missions;
