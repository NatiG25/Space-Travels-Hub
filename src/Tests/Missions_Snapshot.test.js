import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../components/Missions/Missions';
import Mission from '../components/Missions/Mission';
import store from '../components/redux/configureStore';

describe('Test all Missions components and functions', () => {
  it('renders Misssions correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Missions /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Misssion correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Mission /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
