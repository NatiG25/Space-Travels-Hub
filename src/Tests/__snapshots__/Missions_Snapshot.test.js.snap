// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test all Missions components and functions renders Misssion correctly 1`] = `
<tr
  className="mission-row"
>
  <td />
  <td />
  <td
    className="status"
  >
    <span
      className="status-field unavailable-status"
    >
      Not a Member
    </span>
  </td>
  <td>
    <button
      className="button-mission join-mission"
      href="."
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;

exports[`Test all Missions components and functions renders Misssions correctly 1`] = `
<main>
  <table
    className="tableMissions"
  >
    <tbody>
      <tr>
        <th>
          Missions
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </tbody>
  </table>
</main>
`;
